### PR TITLE
Particool 2: Particooler

### PR DESCRIPTION
**Pull request by dwasint**
# About The Pull Request
Fixes uses of spawn, added the ability to add and remove emitters from atoms.
added timer support for burst particles
added new particles from :https://github.com/Baystation12/Baystation12/pull/31947
changed flare to use new particles 

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog

:cl:
update:flares icon was changed to particles
/:cl:

### CODE REVIEW

File: monkestation/code/modules/particles/byond_particles/testing/turf.dm [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/turf/open/particle_test/Initialize(mapload)
+	. = ..()
+	add_emitter(/obj/emitter/fire, "fire")
```
Comment (nednaZ) [inline monkestation/code/modules/particles/byond_particles/testing/turf.dm:3]: Testing turf still needed?

File: monkestation/code/modules/particles/byond_particles/testing/testing.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Any changes to the holder object needs to be tested on turfs as they function differently
```
Comment (nednaZ) [inline monkestation/code/modules/particles/byond_particles/testing/testing.md:1]: Needed?
Could probably roll any minor text like this into particle_usage.md

File: monkestation/code/modules/particles/byond_particles/particle/burst.dm [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	position = generator("box", list(-16, -16), list(16, 16), NORMAL_RAND)
+	velocity = generator("circle", -8, 8, NORMAL_RAND)
+	friction = 0.125
+	color = COLOR_WHITE
+
+/particles/debris
+	width = 124
+	height = 124
+	count = 16
+	spawning = 999999 //spawn all instantly
```
Comment (nednaZ) [inline monkestation/code/modules/particles/byond_particles/particle/burst.dm:17]: May want a define for huge numbers like this, so it looks more readable.
SPAWN_ALL_PARTICLES_INSTANTLY or something

**Review by nednaZ — CHANGES_REQUESTED** [review]
Not seeing anything majorly wrong.

File: code/game/objects/items/devices/flashlight.dm [hunk 4]
```diff
@@ -320,6 +315,8 @@
 		user.visible_message("<span class='notice'>[user] lights \the [src].</span>", "<span class='notice'>You light \the [src]!</span>")
 		force = on_damage
 		damtype = "fire"
+		add_emitter(/obj/emitter/sparks/flare, "spark", 10)
+		add_emitter(/obj/emitter/flare_smoke, "smoke", 9)
```
Comment (nednaZ) [inline code/game/objects/items/devices/flashlight.dm:319]: This incorrectly adds these emitters to the '/obj/item/flashlight/flare/torch' as well, the smoke and sparks are in the wrong location.

Also, the sparks are in the wrong location in general as well.

Comment (dwasint) [reply code/game/objects/items/devices/flashlight.dm:319]: on the torch version? the normal flare looked well on local

File: monkestation/code/modules/particles/byond_particles/testing/turf.dm (hunk 1 above)
Comment (dwasint) [reply monkestation/code/modules/particles/byond_particles/testing/turf.dm:3]: its best to keep in incase we modify stuff in the future

**Review by nednaZ — APPROVED** [review]
Seems fine to me.

**Pull request merged**